termwiz/src/escape/parser/mod.rs

```diff
@@ -81,11 +84,39 @@ impl Parser {
     }
 
     pub fn parse<F: FnMut(Action)>(&mut self, bytes: &[u8], mut callback: F) {
-        let mut perform = Performer {
-            callback: &mut callback,
-            state: &mut self.state.borrow_mut(),
-        };
-        self.state_machine.parse(bytes, &mut perform);
+        let tmux_state: bool = self.state.borrow().tmux_state.is_some();
+        if tmux_state {
+            if let Some(unparsed_str) = {
```

Comment:
This `if let Some(_) = { big block } { // then do something }` construct is hard to read.
Please factor out the `big block` portion into a helper method that returns `Option<String>` so that this section of code is a bit clearer and is focused on routing that unparsed string to the performer.